src/lib.rs

```diff
@@ -210,18 +209,20 @@ impl AstarteSdk {
     /// ```no_run
     /// #[tokio::main]
     /// async fn main() {
-    ///     let mut sdk_options = astarte_sdk::builder::AstarteOptions::new("_","_","_","_")
+    ///     let mut sdk_options = astarte_device_sdk::builder::AstarteOptions::new("_","_","_","_")
     ///                           .build();
-    ///     let mut d = astarte_sdk::AstarteSdk::new(&sdk_options).await.unwrap();
+    ///     let mut d = astarte_device_sdk::AstarteDeviceSdk::new(&sdk_options).await.unwrap();
     ///
     ///     loop {
-    ///         if let Ok(data) = d.poll().await {
+    ///         if let Ok(data) = d.astarte_device_event_handler().await {
     ///             println!("incoming: {:?}", data);
     ///         }
     ///     }
     /// }
     /// ```
-    pub async fn poll(&mut self) -> Result<Clientbound, AstarteError> {
+    pub async fn astarte_device_event_handler(
```

Comment:
I'm not totally convinced by a couple of things:
- Do we need the `astarte_device` prefix? Methods are already namespaced in an `AstarteDeviceSdk` struct
- `event_handler` lets me think of a function _you provide_ to handle event, while here you have to _call_ the function to handle the events, so I'd probably call this `handle_events`, if we want to rename it. This is actually more akin to an eventloop. Alternative names that come to mind: keep `poll`, `tick`, `advance`, `feed`

Reply:
vote for `handle_events`